sfcli/SmartFace.Cli/Commands/SubWatchlistMember/RegisterWatchlistMembersFromDirCmd.cs

```diff
@@ -35,20 +44,48 @@ public class RegisterWatchlistMembersFromDirCmd
         [Option("-p|--parallel", "Max degree of parallelism, default value is 1", CommandOptionType.SingleValue)]
         public int MaxDegreeOfParallelism { get; set; } = 1;
 
-        public RegisterWatchlistMembersFromDirCmd(IWatchlistMemberRegistrationManager registrationManager)
+        public RegisterWatchlistMembersFromDirCmd(IWatchlistMemberRegistrationManager registrationManager, ILogger<RegisterWatchlistMembersFromDirCmd> logger)
         {
             _registrationManager = registrationManager;
+            _logger = logger;
         }
 
-        protected virtual async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole console)
+        public virtual async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole console, CancellationToken cancellationToken)
         {
+            RegistrationResult registrationResult;
+
             if (UseMetaDataFile)
             {
-                await _registrationManager.RegisterWatchlistMembersExtendedFromDirAsync(Directory, WatchlistIds, MaxDegreeOfParallelism);
+                registrationResult = await _registrationManager.RegisterWatchlistMembersFromDirByMetadataFileAsync(Directory, WatchlistIds, MaxDegreeOfParallelism, cancellationToken);
             }
             else
             {
-                await _registrationManager.RegisterWatchlistMembersFromDirAsync(Directory, WatchlistIds, MaxDegreeOfParallelism);
+                registrationResult = await _registrationManager.RegisterWatchlistMembersFromDirAsync(Directory, WatchlistIds, MaxDegreeOfParallelism, cancellationToken);
+            }
+
+            if (registrationResult.Failures.Any())
+            {
+                _logger.LogWarning($"{registrationResult.Failures} registrations failed.");
+
+                var failurePhotosDir = string.IsNullOrEmpty(FailedRegistrationDir)
+                    ? Path.Combine(AppDomain.CurrentDomain.BaseDirectory, Guid.NewGuid().ToString())
```

Comment:
Maybe prepend `Failures` or `Errors`, so that the directory isn't named only by GUID? Also current timestamp may be more informative than a Guid (and unique enough)

Reply:
Updated with prefix and timestamp.